### DISCUSSION TITLE
FormState correct but Fieldstate is not

### DISCUSSION THREAD

**Discussion by SDantas** [react-hook-form/react-hook-form#10077] [Q&A] [answered]
I'm trying to show a message based on the `invalid` flag of `fieldState`.
Why is `invalid` never set even with `required` and `maxLength`?

Even though `formState` is correct the field `errors` is not populated.

Here is the code sandbox - https://codesandbox.io/s/relaxed-mcclintock-ccpnol?file=/src/__tests__/sample.test.tsx
I have simplified the example to use a normal input but I'm using my own input component.

**Comment by leapful** [comment, marked as answer]
@SDantas 

You need to set `mode: 'onChange'` on `useForm` hook to sync validation of isolated field level. Plus, please remember to read `formState.errors` and `formState.isValid` before accessing its values because `formState` is a proxy object that needs to be read in order to sync state.

Working demo: https://codesandbox.io/s/happy-grothendieck-eqe1dq?file=/src/App.tsx

**Reply by SDantas** [reply]
Thanks!
